modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/streaming/AbstractBytesStreamingExtensionTestCase.java

```diff
@@ -348,6 +351,33 @@ public void operationCalledAndOutputConsumedMultipleTimes() throws Exception {
     assertThat(IOUtils.toString((InputStream) value), is(data));
   }
 
+  @Test
+  @Issue("W-16941297")
+  @Description("Call scatter gather containing route to invoke delayed-say-magic-words operation")
+  public void scatterGatherWithTimeout() throws Exception {
+    try {
+      flowRunner("scatterGatherWithTimeout")
+          .keepStreamsOpen()
+          .withPayload(singletonList(data))
+          .run();
+    } catch (Exception e) {
+      assertThat(streamingManager.getStreamingStatistics().getOpenCursorsCount(), is(0));
+    }
```

Comment:
You should use `assertThrows` from JUnit. That way you are making sure the test doesn't succeed if there is no exception.